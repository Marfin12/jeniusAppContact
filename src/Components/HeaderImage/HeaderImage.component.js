import * as React from 'react';
import { TouchableOpacity } from 'react-native';
import { useSelector } from 'react-redux';

import styles from './HeaderImage.styles';
import config from './HeaderImage.config';

const HeaderImage = (props) => {
  const getSelect = useSelector(({ themeReducer }) => themeReducer);
  const { theme } = getSelect;
  const { onPress, image } = props;

  return (
    <TouchableOpacity
      style={[styles.container(theme), style]}
      onPress={onPress(props)}
    >
      <Image source={ image } style={styles.image} />
    </TouchableOpacity>
  );
};

HeaderImage.displayName = config.displayName;
HeaderImage.defaultProps = config.defaultProps;

export default HeaderImage;
