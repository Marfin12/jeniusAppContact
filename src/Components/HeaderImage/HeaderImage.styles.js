import colors from '../../Assets/Colors';

const styles = {
  container: (darkTheme) => ({
    borderColor: darkTheme ? colors.darkHeavyGray : colors.lightHeavyGray,
    borderBottomWidth: 3,
  }),
  image: {
    width: 50,
    height: 50
  }
};

export default styles;
