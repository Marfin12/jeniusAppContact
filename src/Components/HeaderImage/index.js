export { default } from './HeaderImage.component';
