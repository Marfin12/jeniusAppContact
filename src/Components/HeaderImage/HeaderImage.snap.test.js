import React from 'react';
import renderer from 'react-test-renderer';

import HeaderImage from './HeaderImage.component';
import config from './HeaderImage.config';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => darkMode),
  };
});

afterEach(() => {
  jest.resetAllMocks();
});

describe('Header List Component Default Theme', () => {
  const ComponentWrapper = renderer
    .create(<HeaderImage {...config.defaultProps} />)
    .toJSON();

  test('renders header list component light mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});

describe('Header List Component Dark Theme', () => {
  const ComponentWrapper = renderer
    .create(<HeaderImage {...config.defaultProps} />)
    .toJSON();

  test('renders header list component dark mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});
