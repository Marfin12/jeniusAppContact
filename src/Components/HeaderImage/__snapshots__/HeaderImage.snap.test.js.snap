// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header List Component Dark Theme renders header list component dark mode 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomWidth": 3,
      "borderColor": "#222",
      "opacity": 1,
    }
  }
>
  <Image
    source=""
    style={
      Object {
        "height": 36,
        "width": 36,
      }
    }
  />
</View>
`;

exports[`Header List Component Default Theme renders header list component light mode 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomWidth": 3,
      "borderColor": "#EEE",
      "opacity": 1,
    }
  }
>
  <Image
    source=""
    style={
      Object {
        "height": 36,
        "width": 36,
      }
    }
  />
</View>
`;
